tests/test_binary.nim

```diff
@@ -94,6 +94,141 @@ proc testsForSync(binaryPath: string) =
         [warn] some exercises are missing test cases
       """.dedent(8) # Not `unindent`. We want to preserve the indentation of the list items.
 
+    test "a single-exercise `sync` without `--update` exits with 1 and prints the expected output":
+      execAndCheck(1):
+        execCmdEx(&"{binaryPath} -t {trackDir} sync -o -p {psDir} -e anagram")
+
+      check outp == """
+        Checking exercises...
+        [warn] anagram: missing 1 test case
+               - detects two anagrams (03eb9bbe-8906-4ea0-84fa-ffe711b52c8b)
+        [warn] some exercises are missing test cases
+      """.dedent(8)
+
+    test "when passing multiple exercises, only the final exercise is acted upon":
+      # TODO: We should instead support multiple exercises being passed.
```

Comment:
Actually, I don't think we necessarily need to support this. It seems like rather an edge case.

Reply:
Agreed. It's not a priority for me - I'm mainly adding a test case while I'm here to remind myself of the current behavior.

That said, if the user **does** pass multiple `--exercise` arguments, the possible behaviors seem to be:
1. Act on the first one (and probably print a warning)
1. Act on the final one (and probably print a warning)
1. Fail with an error, saying the only one `--exercise` is allowed
1. Act on all of the given exercises

In which case I probably prefer option 4. I think it can sometimes be useful to the maintainer, and is the least surprising and least friction for somebody who does try multiple `--exercise` arguments. And maybe it's useful for scripting.

I don't like the current behavior of acting on just one exercise without a warning. But we can discuss later.